src/core/server/security/readonly_service.ts

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { merge } from 'lodash';
+import { OpenSearchDashboardsRequest, Capabilities } from '../index';
+import { IReadOnlyService } from './types';
+
+export class ReadonlyService implements IReadOnlyService {
+  async isReadonly(request: OpenSearchDashboardsRequest): Promise<boolean> {
```

Comment:
What do you think of changing the output from being boolean to be an enum that has values like `ReadOnly, ReadWrite, Unknown`?

I think this would better describe results for inline reading especially in the controllers where its hard to 'see' what the impact of a true vs false will be.

Reply:
@peternied then it wouldn't be a `ReadonlyService` but `TenantCapabilityService` or something like that. As long as it has an impact only on capabilities I suggest to keep as it is right now and extend that in next PR :)